src/ophyd_async/sim/pattern_generator.py

```diff
@@ -71,9 +63,9 @@ def __init__(
         detector_width: int = 320,
         detector_height: int = 240,
     ) -> None:
-        self.maxshape: tuple[Any, ...]
-        self.fillvalue: int = field(default_factory=int)
-        self.shape: List[int] = field(default_factory=[])
+        self.maxshape: tuple[Any, ...] = (1,)
+        self.fillvalue: int = 1
+        self.shape: List[int] = [1, 1]
```

Comment:
For the next ticket:
- shape of a frame is `[detector_height, detector_width]`
- shape of the DATA HDF dataset is `[1, detector_height, detector_width]`
- fill value for SUM dataset is -1, not required for the dataset
- maxshape for the DATA HDF dataset = `[None, detector_height, detector_width]`

As per https://github.com/dls-controls/bluefly/blob/0ee7ea837d5f74e3cdc69e2315e9beed28dd7766/bluefly/areadetector_sim.py#L55-L68